src/applications/personalization/profile/components/personal-information/ContactInformationEditView.jsx

```diff
@@ -173,36 +256,137 @@ class ContactInformationEditView extends Component {
             <VAPServiceEditModalErrorMessage
               title={title}
               error={error}
-              clearErrors={clearErrors}
+              clearErrors={() => this.props.clearTransactionRequest(fieldName)}
             />
           </div>
         )}
 
         {!!field && (
           <div>
-            {this.props.fieldName === FIELD_NAMES.RESIDENTIAL_ADDRESS && (
+            {fieldName === FIELD_NAMES.RESIDENTIAL_ADDRESS && (
               <CopyMailingAddress
                 copyMailingAddress={this.copyMailingAddress}
               />
             )}
-            <ContactInfoForm
-              formData={field.value}
-              formSchema={field.formSchema}
+
+            <SchemaForm
+              addNameAttribute
+              // `name` and `title` are required by SchemaForm, but are only used
+              // internally by the SchemaForm component
+              name="Contact Info Form"
+              title="Contact Info Form"
+              schema={field.formSchema}
+              data={field.value}
               uiSchema={field.uiSchema}
-              onUpdateFormData={
-                type === 'address'
-                  ? this.onInput
-                  : this.props.onChangeFormDataAndSchemas
-              }
-              onSubmit={onSubmit}
+              onChange={event => {
+                if (addressFieldNames.includes(fieldName)) {
+                  this.onInput(event, field.formSchema, field.uiSchema);
+                  return;
+                }
+
+                this.onChangeFormDataAndSchemas(
+                  event,
+                  field.formSchema,
+                  field.uiSchema,
+                );
+              }}
+              onSubmit={e => onSubmit(e)}
             >
-              {actionButtons}
-            </ContactInfoForm>
+              <ContactInformationActionButtons
+                onCancel={onCancel}
+                onDelete={this.onDelete}
+                title={title}
+                analyticsSectionName={analyticsSectionName}
+                isLoading={isLoading}
+                deleteEnabled={
+                  data && fieldName !== FIELD_NAMES.MAILING_ADDRESS
+                }
+              >
+                <div>
```

Comment:
nit: can this be a Fragment instead?

Reply:
I changed it to a Fragment and it unfortunately messed up the positioning of the buttons, so I reverted to a div.